### PR TITLE
Remove more unnecessary scenarios from HWIntrinsic test templates and fix timeout/failure

### PR DESCRIPTION
**Pull request by tannergooding**
Follow up to https://github.com/dotnet/runtime/pull/85008, this removes the unnecessary load scenarios from the rest of the hwintrinsic test templates.

It also removes the ClsVar scenarios as they weren't actually testing CLS_VAR scenarios. That would require passing in something recognizable as a GT_CNS_VEC or similar, which would require a significantly more complex template or static readonly variables + tiering to kick in.

It also removes the ClassLclFld scenario as there is no real difference between `local._fld` and `this._fld`, the latter being tested by `ClassFld` already.

Finally it divides the original `20` striping used in the `HardwareIntrinsics_r/ro` projects into the new `HardwareIntrinsics_*_r/ro` projects based on the original test count vs the new test count:
* Arm - 2390 generated tests, gets `8` stripes
* General - 2589 generated tests, gets `8` stripes
* X86 - 504 generated tests, gets `2` stripes
* X86_Avx - 500 generated tests, gets `2` stripes
* X86_Avx512 - 200 generated tests (more to be added), gets `2` stripes

----

This also resolves a timeout issue that was cropping up and in turn fixes a downstream failure that was being hidden. It is unclear why this wasn't surfacing on the Linux or Windows x86 legs.

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @BruceForstall, @dotnet/avx512-contrib, @dotnet/jit-contrib 

Also CC. @markples, @trylek for the striping change.

We're still seeing timeouts and long run times for the Avx512 stress job, regardless of striping count. It's possible that is related to https://github.com/dotnet/runtime/issues/84967 and the instruction that is being incorrectly encoded.

**Comment by tannergooding** [comment]
Figured out the timeout issue.

Turns out crossgen was only handling the `X64` nested class and not the `VL` nested class and so `Avx512DQ.VL.Method()` was compiling down to an infinite loop (since it is simply recursive).

This meant that the `RunReflectionScenario_UnsafeRead` scenario would hang the tests indefinitely.

File: src/coreclr/jit/hwintrinsiclistxarch.h [hunk 1]
```diff
@@ -801,7 +801,7 @@ HARDWARE_INTRINSIC(AVX512F,         ConvertToVector128UInt16,
...
 HARDWARE_INTRINSIC(AVX512F,         ConvertToVector256Int32,                    64,              1,     {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_vpmovqd,            INS_vpmovqd,            INS_invalid,            INS_cvtpd2dq},          HW_Category_SimpleSIMD,             HW_Flag_BaseTypeFromFirstArg|HW_Flag_SpecialCodeGen)
-HARDWARE_INTRINSIC(AVX512F,         ConvertToVector128Int32WithTruncation,      64,              1,     {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_cvttpd2dq},         HW_Category_SimpleSIMD,             HW_Flag_BaseTypeFromFirstArg)
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsiclistxarch.h:804]: so we don't need `ConvertToVector128Int32WithTruncation` at all for AVX512F?

**Comment by kunalspathak** [comment]
> was compiling down to an infinite loop (since it is simply recursive)

Interestingly we didn't find out in the initial PR CI?

**Comment by tannergooding** [comment]
> Interestingly we didn't find out in the initial PR CI?

It was dependent on a couple few factors, such as whether or not the R2R assembly is used. It was unclear why Linux nor Windows x86 failed and why it was only showing up on Windows x64 in CI.

File: src/coreclr/jit/hwintrinsiccodegenxarch.cpp [hunk 2]
```diff
@@ -1786,12 +1786,22 @@ void CodeGen::genAvxFamilyIntrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_AVX512F_ConvertToVector256Int32:
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsiccodegenxarch.cpp:1789]: why was this moved up?

Comment (tannergooding) [reply src/coreclr/jit/hwintrinsiccodegenxarch.cpp:1789]: To be consistent with the other places that specially handle this one path and to avoid the cost of the `varTypeIsFloating` check for the many intrinsics where it can't be true.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Review by EgorBo — APPROVED** [review]

**Comment by tannergooding** [comment]
Merging after talking with @BruceForstall.

The only jobs that timed out are the arm32 replay jobs, which won't be impacted by this PR as we're only touching xarch specific files/paths. The reply job did pass earlier today before the minor test fix went in.

**Review by BruceForstall — APPROVED** [review]
